device.c

```diff
@@ -448,7 +448,18 @@ void cSatipDevice::CloseDvr(void)
 bool cSatipDevice::HasLock(int timeoutMsP) const
 {
   debug16("%s (%d) [device %d]", __PRETTY_FUNCTION__, timeoutMsP, deviceIndexM);
-  return (pTunerM && pTunerM->HasLock());
+  if (timeoutMsP <= 0)
+     return (pTunerM && pTunerM->HasLock());
+  else {
+     const int intervalMs = 100;
+
+     for (int t = 0; t < timeoutMsP; t += intervalMs) {
+         if (HasLock(0))
+            return true;
+         cCondWait::SleepMs(intervalMs);
+         }
+     return HasLock(0);
```

Comment:
Why avoid recursion, if its better solution here?
If using  `pTunerM->HasLock()`,
we actually need to use `(pTunerM && pTunerM->HasLock())` instead.

Reply:
Could you pls respond, i m waiting for you. Would reduce greatly efforts for both of us.

Do you prefer
1) 
bool cSatipDevice::HasLock(int timeoutMsP) const
{
  debug16("%s (%d) [device %d]", __PRETTY_FUNCTION__, timeoutMsP, deviceIndexM);
  if (timeoutMsP <= 0)
     return (pTunerM && pTunerM->HasLock());

  cTimeMs timer(timeoutMsP);

  while (!timer.TimedOut()) {
        if (HasLock(0))
           return true;
        cCondWait::SleepMs(100);
        }
  return HasLock(0);
}


**or**

2)
bool cSatipDevice::HasLock(int timeoutMsP) const
{
  debug16("%s (%d) [device %d]", __PRETTY_FUNCTION__, timeoutMsP, deviceIndexM);
  if (timeoutMsP <= 0)
     return (pTunerM && pTunerM->HasLock());

  cTimeMs timer(timeoutMsP);

  while (!timer.TimedOut()) {
        if (pTunerM && pTunerM->HasLock())
           return true;
        cCondWait::SleepMs(100);
        }
  return (pTunerM && pTunerM->HasLock());
}

Reply:
Sorry for the delay. My intention to avoid recursion here is to possibility to add a mutex here. Let's go forward with modified version 2:
```
bool cSatipDevice::HasLock(int timeoutMsP) const
{
    debug16("%s (%d) [device %d]", PRETTY_FUNCTION, timeoutMsP, deviceIndexM);
    if (timeoutMsP > 0) {
        cTimeMs timer(timeoutMsP);
        while (!timer.TimedOut()) {
            if (pTunerM && pTunerM->HasLock())
                 return true;
             cCondWait::SleepMs(100);
             }
        }
    return (pTunerM && pTunerM->HasLock());
}
````

Reply:
done -> v3